core/src/apps/cardano/sign_tx.py

```diff
@@ -100,526 +111,529 @@
 
 @seed.with_keychain
 async def sign_tx(
-    ctx: wire.Context, msg: CardanoSignTx, keychain: seed.Keychain
-) -> CardanoSignedTx:
-    if msg.fee > LOVELACE_MAX_SUPPLY:
-        raise wire.ProcessError("Fee is out of range!")
+    ctx: wire.Context, msg: CardanoSignTxInit, keychain: seed.Keychain
+) -> CardanoSignTxFinished:
+    is_network_id_verifiable = await _validate_tx_signing_request(ctx, msg)
+
+    # inputs, outputs and fee are mandatory fields, count the number of optional fields present
+    tx_body_map_item_count = 3 + sum(
+        1
+        for entry in [
+            msg.ttl,
+            msg.certificates_count,
+            msg.withdrawals_count,
+            msg.has_auxiliary_data,
+            msg.validity_interval_start,
+        ]
+        if bool(entry)
```

Comment:
I think it's because it's a nice way to convert all the values in the list (optional ints, ints and a bool) into bool. However it contains a bug which causes tx signing to fail if `ttl` or `validity_interval_start` are zero. So I updated it and also added a test for `ttl` and `validity_interval_start` being zero.